docs/base-account/framework-integrations/privy/spend-permissions.mdx

```diff
@@ -0,0 +1,289 @@
+      
+      if (!isActive) {
+        console.error("Selected permission is not active");
+        return;
+      }
+
+      const spendAmount = BigInt(100) * BigInt(10 ** 6); // 100 USDC
+      
+      if (remainingSpend < spendAmount) {
+        console.error("Insufficient remaining allowance");
+        return;
+      }
+
+      // Prepare spend calls
+      const spendCalls = await prepareSpendCallData(selectedPermission, spendAmount);
+      console.log("Spend calls prepared:", spendCalls);
+      
+    } catch (error) {
+      console.error("Failed to use Spend Permission:", error);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  return (
+    <div className="space-y-4">
+      <div className="flex gap-4">
+        <button 
+          onClick={handleRequestSpendPermission}
+          disabled={loading || !account}
+          className="px-4 py-2 bg-blue-600 text-white rounded disabled:opacity-50"
+        >
+          Create Spend Permission
+        </button>
+        <button 
+          onClick={loadPermissions}
+          disabled={loading || !account}
+          className="px-4 py-2 bg-green-600 text-white rounded disabled:opacity-50"
+        >
+          Load Permissions
+        </button>
+        <button 
+          onClick={handleUseSpendPermission}
+          disabled={loading || !selectedPermission}
+          className="px-4 py-2 bg-purple-600 text-white rounded disabled:opacity-50"
+        >
+          Use Permission
+        </button>
+      </div>
+
+      {/* Configuration display */}
+      <div className="bg-gray-50 p-4 rounded-lg">
+        <h4 className="font-semibold mb-3">Configuration</h4>
+        <div className="text-sm space-y-1">
+          <div><strong>Spender:</strong> {spenderAddress}</div>
+          <div><strong>Token:</strong> {tokenAddress} (USDC)</div>
+          <div><strong>Allowance:</strong> $1 USDC per day</div>
+        </div>
+      </div>
+
+      {/* Permissions list */}
+      {permissions.length > 0 && (
+        <div className="bg-gray-50 p-4 rounded-lg">
+          <h4 className="font-semibold mb-3">Existing Permissions</h4>
+          <div className="space-y-2">
+            {permissions.map((permission, index) => (
+              <div
+                key={index}
+                className={`p-3 border rounded-md cursor-pointer ${
+                  selectedPermission === permission
+                    ? "border-blue-500 bg-blue-50"
+                    : "border-gray-300"
+                }`}
+                onClick={() => setSelectedPermission(permission)}
+              >
+                <div className="text-sm">
+                  <div><strong>Spender:</strong> {permission.spender?.slice(0, 10)}...</div>
+                  <div><strong>Token:</strong> {permission.token?.slice(0, 10)}...</div>
+                  <div><strong>Allowance:</strong> {permission.allowance?.toString()}</div>
```

Comment:
maybe consider renaming permission to `spendPermission` so that we avoid the `permission.permission`
```suggestion
                  <div><strong>Spender:</strong> {permission.permission.spender?.slice(0, 10)}...</div>
                  <div><strong>Token:</strong> {permission.permission.token?.slice(0, 10)}...</div>
                  <div><strong>Allowance:</strong> {permission.permission.allowance?.toString()}</div>
```